charmcraft.yaml

```diff
@@ -12,5 +12,6 @@ bases:
       channel: "20.04"
 parts:
   charm:
+    build-packages: [cargo, rustc, pkg-config, libffi-dev, libssl-dev]
```

Comment:
I think you can remove `cargo`, `rust` and `pkg-config` by including
```
build-snaps:
      - rustup
```
instead

Reply:
oh cool, let me try that. Thanks!

Reply:
That did not work actually.

`error: can't find Rust compiler`
log:
https://pastebin.canonical.com/p/hjj7kVzzs8/

Reply:
rustc and cargo should be included, as per https://www.rust-lang.org/tools/install. Can you try keeping pkg-config and cleaning the charmcraft cache just in case?

Reply:
Sure, trying it out

Reply:
Tried:
```python

  charm:
    build-packages: [cargo, rustc, pkg-config, libffi-dev, libssl-dev]
    build-snaps:
      - rustup
```
and 
```python

  charm:
    build-packages: [cargo, rustc, pkg-config]
    build-snaps:
      - rustup
```

Both did not work.

Reply:
Well, let's keep it as it is then